lib/index.js

```diff
@@ -225,15 +225,19 @@ Validator.prototype._setup = function () {
 
   if(this.origin) this._addFromOrigin();
 
-  this.model.on('change', '**.value', function (field, value) {
-    if (self.origin) {
-      if (value !== self.origin.get(field)) {
-        self.model.set(field + '.hasChanged', true);
-      } else {
-        self.model.set(field + '.hasChanged', false);
-      }
+  this.model.on('change', '**', function (path, value) {
+    var segments = path.split('.');
+    if (_.last(segments) === 'value') {
+      var field = _.dropRight(segments).join('.');
+      if (self.origin) {
```

Comment:
Why only for origin? Shouldn't one be able to track also for non-origin validator instances, where for example one would like to check whenever somebody entered anything (compared to if somebody changed the origin value)?